pco/src/latent_batch_dissector.rs

```diff
@@ -42,16 +49,33 @@ impl<'a, L: Latent> LatentBatchDissector<'a, L> {
       }
     }
 
-    let max_idx = self.table.infos.len().saturating_sub(1);
-    search_idxs
-      .iter_mut()
-      .for_each(|search_idx| *search_idx = min(*search_idx, max_idx));
+    let n_bins = self.table.infos.len();
+    if n_bins < 1 << self.table.search_size_log {
+      // We worked with a balanced binary tree with missing leaves filled, so it
+      // might have overshot some bin indices.
+      search_idxs
+        .iter_mut()
+        .for_each(|search_idx| *search_idx = min(*search_idx, n_bins - 1));
+    }
 
     search_idxs
   }
 
   #[inline(never)]
   fn dissect_bins(&mut self, search_idxs: &[usize], dst_offset_bits: &mut [Bitlen]) {
+    if self.table.is_trivial() {
+      // trivial case: there's at most one bin. We've prepopulated the scratch
+      // buffers with the correct values in this case.
```

Comment:
Both scratch buffers?  This might be as simple as "if there is only one symbol, then the symbol is 0," but just in case I'm misunderstanding, as only the `lower_scratch` buffer is mentioned above.

Reply:
Yes, both buffers. I added to the comment above to make this clearer.